.gitignore

```diff
@@ -0,0 +1,7 @@
+*.o
```

Comment:
We have traditionally not included a '.gitignore' file in the repo.  It looks like many argue that it should be included.  Opinions?

Reply:
When you add new files, it is very easy to miss them when you do a git status and it is 2 pages long. With a good .gitignore only the files that matter will show up.

Reply:
I generally don't like it in the repo, since I have my own in home directory that includes particular ignored files such as vi swap files. Having this one will override it. Just my preference.

Reply:
I also have one in my home directory, but there is a decent argument for including one in the repo as well.  Especially for weird non-standard files that the library generates (we have our '.obj' files, for example), or files that you want to ensure people don't accidentally commit.

Reply:
Having both means a union of the local and global sets will be applied right ?

Reply:
Not sure yet.  There are some precedence rules, but I need to read this again: https://git-scm.com/docs/gitignore

Reply:
Okay, I just did some more reading on this.  It appears that a union of gitignore patterns will happen in most cases, as speculated by @Ram3sh , but there is a prefix '!' that negates (un-ignores) a previous pattern, so having a precedence rule for the different gitignore files becomes important.  The local file takes precedence.

We could also add the following to this

    *.so
    config.log
    config.status

Given this, what do you think @liruipeng ?

Reply:
@rfalgout Thank you for researching this. Given this and the fact that I don't have any negated patterns so far in my local one, I think it's safe to have a global hypre's `.gitignore` (at least for me). The addings you want to make look good to me. Thanks!

Reply:
FYI, I just confirmed this.  We probably should also add the following, since we don't want them commited accidentally:

    Makefile.config
    HYPRE_config.h

Reply:
I will make the change.  Thanks!

Reply:
@rfalgout there's also a autom4te.cache folder that is generated if the (autconf) build system is modified and a new configure file is generated. Should this make it into the .gitignore file as well?

Reply:
Just added it.  I had it in my global gitignore file, but I couldn't remember what it was for.  Thanks!